src/libs/Navigation/Navigation.ts

```diff
@@ -49,37 +44,32 @@ const getTopmostReportId = (state = navigationRef.getState()) => originalGetTopm
 // Re-exporting the getTopmostReportActionID here to fill in default value for state. The getTopmostReportActionID isn't defined in this file to avoid cyclic dependencies.
 const getTopmostReportActionId = (state = navigationRef.getState()) => originalGetTopmostReportActionId(state);
 
-/**
- * Method for finding on which index in stack we are.
- * @param {Object} route
- * @param {Number} index
- * @returns {Number}
- */
-const getActiveRouteIndex = function (route, index) {
-    if (route.routes) {
-        const childActiveRoute = route.routes[route.index || 0];
-        return getActiveRouteIndex(childActiveRoute, route.index || 0);
+/** Method for finding on which index in stack we are. */
+function getActiveRouteIndex(stateOrRoute: StateOrRoute, index?: number): number | undefined {
+    if ('routes' in stateOrRoute && stateOrRoute.routes) {
+        const childActiveRoute = stateOrRoute.routes[stateOrRoute.index ?? 0];
+        return getActiveRouteIndex(childActiveRoute, stateOrRoute.index ?? 0);
     }
 
-    if (route.state && route.state.routes) {
-        const childActiveRoute = route.state.routes[route.state.index || 0];
-        return getActiveRouteIndex(childActiveRoute, route.state.index || 0);
+    if ('state' in stateOrRoute && stateOrRoute?.state?.routes) {
+        const childActiveRoute = stateOrRoute.state.routes[stateOrRoute.state.index ?? 0];
+        return getActiveRouteIndex(childActiveRoute, stateOrRoute.state.index ?? 0);
     }
 
-    if (route.name === NAVIGATORS.RIGHT_MODAL_NAVIGATOR) {
+    if ('name' in stateOrRoute && stateOrRoute.name === NAVIGATORS.RIGHT_MODAL_NAVIGATOR) {
```

Comment:
There is no guideline for this, this is needed for [type narrowing](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#the-in-operator-narrowing). stateOrRoute is a union of couple types, without `'name' in stateOrRoute` we get TS error `Property 'name' does not exist on type 'StateOrRoute'`

Reply:
@blazejkustra just curios why we can't use `stateOrRoute?.name === NAVIGATORS.RIGHT_MODAL_NAVIGATOR`

Reply:
`stateOrRoute` cannot be null if you deep look into code.
Indeed, it causes lint error. So I'd say this is just to fix lint rather than "intentional"

<img width="931" alt="Screenshot 2023-11-24" src="https://github.com/Expensify/App/assets/108292595/a3617acd-9311-41e6-84fc-16453163f998">

Reply:
> @blazejkustra just curios why we can't use `stateOrRoute?.name === NAVIGATORS.RIGHT_MODAL_NAVIGATOR`

At this point in the function, we have to prove that `stateOrRoute` has `name` prop inside (we narrow down the type with `in` operator)

> Indeed, it causes lint error. So I'd say this is just to fix lint rather than "intentional"

It's not a lint error, it is a typescript error @situchan 😄

Reply:
it makes sense, thanks @situchan @blazejkustra